### DISCUSSION TITLE
mitmproxy 9.0.1版本中 ProxyConfig和ProxyServer 用什么替代了

### DISCUSSION THREAD

**Discussion by yaomoshen5200** [mitmproxy/mitmproxy#6061] [Q&A]
在mitmproxy 6.0.0 中有from mitmproxy.proxy.config import ProxyConfig
from mitmproxy.proxy.server import ProxyServer 这两个类 但是在9.0.1 中没有了 那么在9.0.1中如何实现python代码启动 mitmproxy呢
在6.0.0中
  options = Options(listen_host='0.0.0.0', listen_port=8080)
    master = DumpMaster(options=options,with_termlog=False, with_dumper=False)
    config = ProxyConfig(options)
    master.server = ProxyServer(config)
    master.addons.add(Addon())
    master.run()
这么操作的 那么在9.0.1中如何操作

**Comment by sujaldev** [comment]
I don't know chinese so there's a chance I might've misread something. Not sure if this is the _right_ way to do this, but it'll work:

```python
from mitmproxy.tools.main import mitmproxy #, mitmweb, mitmdump

mitmproxy()  # mitmproxy(["some", "argument"])
```

This test is using the same approach: https://github.com/mitmproxy/mitmproxy/blob/be02b1e298dfb1b218007f824c4cf345c53b0cee/test/mitmproxy/addons/test_errorcheck.py#L12-L17

---

AI translated (AI 翻译):

我不懂中文，所以可能会误读某些内容。不确定这是否是“正确”的方式，但这会起作用：

```python
from mitmproxy.tools.main import mitmproxy #, mitmweb, mitmdump

mitmproxy()  # mitmproxy(["some", "argument"])
```

这个测试使用了相同的方法：https://github.com/mitmproxy/mitmproxy/blob/be02b1e298dfb1b218007f824c4cf345c53b0cee/test/mitmproxy/addons/test_errorcheck.py#L12-L17

**Reply by yaomoshen5200** [reply]
What I mean is that 
options = Options(listen_host='0.0.0.0', listen_port=8080)
master = DumpMaster(options=options,with_termlog=False, with_dumper=False)
config = ProxyConfig(options)
master.server = ProxyServer(config)
master.addons.add(Addon())
master.run()
 can be packaged as exe in version 6.0.0, so how should I write it in version 9.0.1 to be packaged as exe

**Reply by sujaldev** [reply]
Well if you know English why'd you write it in Chinese? Anyways, what are you using to build the exe, pyinstaller? If yes, the above approach is working for me.

**Reply by yaomoshen5200** [reply]
I didn't know English, but I translated Chinese into English
